pkg/kubelet/collector/types.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package collector
+
+import (
+	"time"
+)
+
+// TODO: These structs are mostly taken directly from cadvisor to minimize the
+// amount of changes we need to do in this iteration, but these eventually need
+// to be cleaned up further to be less cadvisor-specific and more generic to kubernetes
+
+// Basic node information
+type NodeInfo struct {
+	// The number of cores on this node
+	NumCores uint `json:"num_cores"`
```

Comment:
- json should be camelCase
- avoid unsigned integer types
- need to specify int size (should be `int32`)
- optional fields should be pointer type, and labeled with `omitempty` (I'm not sure which of these are optional, but it's worth thinking about)